### ISSUE TITLE
Copying a wrapped line does not work as it should

### ISSUE THREAD

**Issue by mingwandroid** [sedwards2009/extraterm#69]
If you copy a very long line that has been wrapped over multiple lines and paste it again, it is pasted as multiple lines split where the wrapping occurred.

**Comment by sedwards2009** [comment]
This feature is closely related to general feature of "word wrap on resize".

**Comment by marksandler2** [comment]
so, what's the recommended way of copy pasting item to avoid spurious newlines (suppose i am not resizing window). This is particularly important when i am copying/pasting commands.

**Comment by sedwards2009** [comment]
The closest thing to recommended way is to paste your lines into a text editor manually fix the lines, select and copy again. Yes I know it is not a great solution.

**Comment by marksandler2** [comment]
Thanks, yeah, thats not very useful indeed. :-) So looking at the dev panel it appears that extraterm uses codemirror for rendering outputs - code mirror support line word wrapping natively, (https://codemirror.net/demo/indentwrap.html) 

So if  extraterm can feed complete lines to codemirror, then the right thing should automatically happen.

**Comment by sedwards2009** [comment]
Getting codemirror to  do wrapping for lines shouldn't be super hard for lines which have entered the scrollback. For lines which are still on the screen and can be modified via terminal escape codes, the situation is trickier.

**Comment by marksandler2** [comment]
My preference is for terminal to treat long lines that wrapped over as a single line from the perspective of terminal escape codes, rather than multiple lines.  E.g. term-> go to the beginning of the line  should go to the  real the beginning of the line (that is where it would go if terminal had infinite width), not to the same on-screen line. I believe this is also the expectation for most of applications but i could be wrong. 

If you are open to this, i can see if i can fix this with minimal blood though pointers are appreciated.

**Comment by sedwards2009** [comment]
I have a feeling that changing how the cursor behaves when it reaches the end of a row is a bad idea. I strongly suspect that some 'rich' terminal based UIs expect the cursor to appear on the next line. `vttest` already shows that Extraterm lacks a bit in its emulation already.

If you want to investigate this a bit more, then it makes sense to look in `Term.ts`, get a feel for how it works and think about how to flag rows as "wrapped" when the cursor runs off the end. This info could then be passed on to the rendering stage which could use it to 'correct' text when it is copied.

**Comment by psprint** [comment]
I'm having the same problem: *copying/pasting long, wrapped lines* **––>**  *copying multiple lines* (should be: *copying as single line, unless an actual `\n` occurs in the selected text*)

**Comment by zicklag** [comment]
:+1: for this. It can be pretty inconvenient when I want to copy output from Extraterm if it wraps.

**Comment by nutcr4cker** [comment]
keep finger crossed for that improvement as its a bit annoying

**Comment by accountgit** [comment]
Tried in current release and continues happening. 
Will be a VERY welcome feature.

**Comment by maroony** [comment]
Please fix this. Every other terminal emulator behaves that way.

**Comment by sedwards2009** [comment]
Released in v0.45.0.

**Comment by maroony** [comment]
The problem still exists when you try to copy & paste a line within extraterm:

```
$ echo A very long ... 
line.
```

Now  copy & paste this command and you get
```
$ echo A very long ...
$ line.
```

What you should get is 

```
$ echo A very long ...
line.
```
